### PR TITLE
Ez editor chain switching

### PR DESCRIPTION
**Pull request by sneakzttv**
We now have the ability to easily switch chains in the editor via drop down menus in the server settings area! Woohoo!
The values are populated dynamically from Chainlist's public github so we're always going to be up to date with new chains.

image

------------------------To Test---------------------------
Boot up the SDK as usual and go to the server settings area.
Ensure the search functionality functions as intended, you should only see results relevant to the current text input.
Select a chain plus an RPC from the drop down menus and press save.
Ensure the network.js files are updating with this information.
Test that server information is saving between window and editor restarts.
Test that some sample calls are working for standard & webgl builds (I didn't change anything here, just a quick sanity check)

Expected behaviors:
Adding a custom RPC will override the drop down menus selected RPC.
When a custom RPC set and you change chains or RPCs from the menus, it should remove the custom RPC value automatically.
Network, chainID & symbol values should all be updating dynamically as you change chains.

### CODE REVIEW

File: Packages/io.chainsafe.web3-unity/Editor/ChainInfoModel.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+using System.Collections.Generic;
+
+namespace ChainSafe.Gaming.UnityPackage.Model
```
Comment (robGG1997) [inline Packages/io.chainsafe.web3-unity/Editor/ChainInfoModel.cs:3]: 3 suggestions for this class
1. some of these can be structs except the main class
2. if we're deserializing the response we can just have the fields we need (chain name and id)
3. Model in namespace should be replaced with editor or move script into Runtime/Scripts/Model like all other models

File: Packages/io.chainsafe.web3-unity/Editor/ServerSettings.cs [hunk 2]
```diff
@@ -1,46 +1,119 @@
...
     private string symbol;
     private string rpc;
+    private string newRpc;
 
     Texture2D logo = null;
+    
+    // Search window
+    private StringListSearchProvider searchProvider;
+    private ISearchWindowProvider _searchWindowProviderImplementation;
+    public static UnityEvent onDropDownChange;
```
Comment (robGG1997) [inline Packages/io.chainsafe.web3-unity/Editor/ServerSettings.cs:46]: can we reference this event from the search provider?

**Review by robGG1997 — CHANGES_REQUESTED** [review]

File: Packages/io.chainsafe.web3-unity/Editor/ServerSettings.cs (hunk 2 above)
Comment (sneakzttv) [reply Packages/io.chainsafe.web3-unity/Editor/ServerSettings.cs:46]: done

File: Packages/io.chainsafe.web3-unity/Editor/ChainInfoModel.cs (hunk 1 above)
Comment (sneakzttv) [reply Packages/io.chainsafe.web3-unity/Editor/ChainInfoModel.cs:3]: done

File: Packages/io.chainsafe.web3-unity/Editor/StringListSearchProvider.cs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        onSetIndexCallback?.Invoke((string)SearchTreeEntry.userData);
+        ChainSafeServerSettings instance = EditorWindow.GetWindow<ChainSafeServerSettings>();
+        if (instance != null)
+        {
+            instance.UpdateServerMenuInfo();
+        }
+        else
+        {
+            // Throw an exception when instance is null
+            throw new NullReferenceException("ChainSafeServerSettings instance is null. Cannot update server menu info.");
```
Comment (robGG1997) [inline Packages/io.chainsafe.web3-unity/Editor/StringListSearchProvider.cs:105]: if you're throwing a null ref exception you can just `instance.UpdateServerMenuInfo();` without checking if instance is null

Comment (sneakzttv) [reply Packages/io.chainsafe.web3-unity/Editor/StringListSearchProvider.cs]: done

**Review by kantagara — APPROVED** [review]

**Review by robGG1997 — APPROVED** [review]

**Review by juans-chainsafe — APPROVED** [review]
Looks good! just a minor issue that I will create a ticket, but is not a blocker.

**Pull request merged**